--- a/rust-vendor/aya/Cargo.toml
+++ b/rust-vendor/aya/Cargo.toml
@@ -11,7 +11,7 @@
 
 [package]
 edition = "2021"
-rust-version = "1.80.0"
+rust-version = "1.66"
 name = "aya"
 version = "0.13.1"
 authors = ["Aya Contributors"]
@@ -67,6 +67,10 @@
 version = "1"
 default-features = false
 
+[dependencies.lazy_static]
+version = "1"
+default-features = false
+
 [dependencies.libc]
 version = "0.2.105"
 default-features = false
--- a/rust-vendor/aya/src/bpf.rs
+++ b/rust-vendor/aya/src/bpf.rs
@@ -7,7 +7,7 @@
         raw::c_int,
     },
     path::{Path, PathBuf},
-    sync::{Arc, LazyLock},
+    sync::Arc,
 };
 
 use aya_obj::{
@@ -70,7 +70,9 @@
 
 pub use aya_obj::maps::{bpf_map_def, PinningType};
 
-pub(crate) static FEATURES: LazyLock<Features> = LazyLock::new(detect_features);
+lazy_static::lazy_static! {
+    pub(crate) static ref FEATURES: Features = detect_features();
+}
 
 fn detect_features() -> Features {
     let btf = if is_btf_supported() {
--- a/rust-vendor/aya/src/programs/uprobe.rs
+++ b/rust-vendor/aya/src/programs/uprobe.rs
@@ -8,7 +8,7 @@
     mem,
     os::{fd::AsFd as _, raw::c_char, unix::ffi::OsStrExt},
     path::{Path, PathBuf},
-    sync::LazyLock,
+    sync::Arc,
 };
 
 use libc::pid_t;
@@ -29,8 +29,10 @@
 
 const LD_SO_CACHE_FILE: &str = "/etc/ld.so.cache";
 
-static LD_SO_CACHE: LazyLock<Result<LdSoCache, io::Error>> =
-    LazyLock::new(|| LdSoCache::load(LD_SO_CACHE_FILE));
+lazy_static::lazy_static! {
+    static ref LD_SO_CACHE: Result<LdSoCache, Arc<io::Error>> =
+        LdSoCache::load(LD_SO_CACHE_FILE).map_err(Arc::new);
+}
 const LD_SO_CACHE_HEADER_OLD: &str = "ld.so-1.7.0\0";
 const LD_SO_CACHE_HEADER_NEW: &str = "glibc-ld.so.cache1.1";
 
--- a/rust-vendor/aya/src/programs/utils.rs
+++ b/rust-vendor/aya/src/programs/utils.rs
@@ -5,8 +5,7 @@
     io::{self, BufRead, BufReader},
     os::fd::{AsFd as _, AsRawFd as _, BorrowedFd},
     path::Path,
-    sync::LazyLock,
-    time::{Duration, SystemTime, UNIX_EPOCH},
+    time::{Duration, SystemTime, UNIX_EPOCH}
 };
 
 use crate::{
@@ -32,26 +31,27 @@
 
 /// Find tracefs filesystem path.
 pub(crate) fn find_tracefs_path() -> Result<&'static Path, ProgramError> {
-    static TRACE_FS: LazyLock<Option<&'static Path>> = LazyLock::new(|| {
-        [
-            Path::new("/sys/kernel/tracing"),
-            Path::new("/sys/kernel/debug/tracing"),
-        ]
-        .into_iter()
-        .find(|&mount| {
-            // Check that the mount point exists and is not empty
-            // Documented here: (https://www.kernel.org/doc/Documentation/trace/ftrace.txt)
-            // In some cases, tracefs will only mount at /sys/kernel/debug/tracing
-            // but, the kernel will still create the directory /sys/kernel/tracing.
-            // The user may be expected to manually mount the directory in order for it to
-            // exist in /sys/kernel/tracing according to the documentation.
-            mount.exists()
-                && match mount.read_dir() {
-                    Ok(mut entries) => entries.next().is_some(),
-                    Err(io::Error { .. }) => false,
+    lazy_static::lazy_static! {
+        static ref TRACE_FS: Option<&'static Path> = {
+            let known_mounts = [
+                Path::new("/sys/kernel/tracing"),
+                Path::new("/sys/kernel/debug/tracing"),
+            ];
+
+            for mount in known_mounts {
+                // Check that the mount point exists and is not empty
+                // Documented here: (https://www.kernel.org/doc/Documentation/trace/ftrace.txt)
+                // In some cases, tracefs will only mount at /sys/kernel/debug/tracing
+                // but, the kernel will still create the directory /sys/kernel/tracing.
+                // The user may be expected to manually mount the directory in order for it to
+                // exist in /sys/kernel/tracing according to the documentation.
+                if mount.exists() && mount.read_dir().ok()?.next().is_some() {
+                    return Some(mount);
                 }
-        })
-    });
+            }
+            None
+        };
+    }
 
     TRACE_FS
         .as_deref()
